contracts/metagov/BalancerGaugeStaker.sol

```diff
@@ -0,0 +1,91 @@
+        CoreRef._initialize(_core);
+        gaugeController = BALANCER_GAUGE_CONTROLLER;
+    }
+
+    /// @notice returns total balance of PCV in the Deposit
+    function balance() public view override returns (uint256) {
+        return IERC20(BAL).balanceOf(address(this));
+    }
+
+    /// @notice gets the token address in which this deposit returns its balance
+    function balanceReportedIn() public view virtual override returns (address) {
+        return BAL;
+    }
+
+    /// @notice gets the resistant token balance and protocol owned fei of this deposit
+    function resistantBalanceAndFei() public view virtual override returns (uint256, uint256) {
+        return (balance(), 0);
+    }
+
+    /// @notice noop
+    function deposit() external override {}
+
+    /// @notice withdraw BAL held to another address
+    /// the BAL rewards accrue on this PCVDeposit when Gauge rewards are claimed.
+    function withdraw(address to, uint256 amount) external override onlyPCVController whenNotPaused {
+        IERC20(BAL).safeTransfer(to, amount);
+        emit Withdrawal(msg.sender, to, amount);
+    }
+
+    /// @notice withdraw ERC20 from the contract
+    /// @param token address of the ERC20 to send
+    /// @param to address destination of the ERC20
+    /// @param amount quantity of ERC20 to send
+    function withdrawERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) public virtual override onlyPCVController whenNotPaused {
+        // if the token has a gauge, we unstake it from the gauge before transfer
+        // @dev : note that this gauge unstaking normally is access controlled to
+        // onlyTribeRole(TribeRoles.METAGOVERNANCE_GAUGE_ADMIN), but since
+        // withdrawERC20() is a higher clearance role (PCVController), this
+        // privilege of gauge unstaking is also made available to the PCV controller,
+        // that may have important reasons to quickly unstake from the gauge and move
+        // the staked ERC20s to another place.
+        address gaugeAddress = tokenToGauge[token];
+        if (gaugeAddress != address(0)) {
+            ILiquidityGauge(gaugeAddress).withdraw(amount, false);
+        }
+
+        _withdrawERC20(token, to, amount);
+    }
+
+    /// @notice Mint everything which belongs to this contract in the given gauge
+    /// @param token whose gauge should be claimed
+    function mint(address token) external whenNotPaused returns (uint256) {
+        // fetch gauge address from internal mapping to avoid this permissionless
+        // call to mint on any arbitrary gauge.
+        address gaugeAddress = tokenToGauge[token];
+        require(gaugeAddress != address(0), "BalancerGaugeStaker: token has no gauge configured");
+
+        // emit the same event as Gauge rewards claiming for offline indexing
+        uint256 minted = IBalancerMinter(BALANCER_MINTER).mint(gaugeAddress);
+        emit GaugeRewardsClaimed(gaugeAddress, BAL, minted);
```

Comment:
what about the backwards compatible rewards logic for second tokens like LDO

Reply:
the LiquidityGaugeManager class inherited has gauge claiming functions (which distribute LDO and the likes)